lib/getDigest.js

```diff
@@ -0,0 +1,41 @@
+const crypto = require('crypto')
+
+const getDigest = (
+  {
+    headers: {
+      digest,
+      'content-type': contentType,
+      'content-encoding': contentEncoding
+    },
+    body
+  }, options
+) => {
+  const { formattedAlgorithm, digestAlgorithm, digestValue } = parseBodyDigestHeader(digest)
+
+  const calculatedDigest = calculateDigest({
+    algorithm: formattedAlgorithm,
+    message: JSON.stringify(body),
+    encoding: options.digestEncoding
+  })
+
+  if (calculatedDigest !== digestValue) {
+    throw new Error('Calculated Digest does not match Digest from Request Header')
+  }
+
+  return `${digestAlgorithm}=${calculatedDigest}`
+}
+
+const parseBodyDigestHeader = (requestDigestHeader) => {
+  const [digestAlgorithm, digestValue] = requestDigestHeader.split(/=(.+)/)
+  const formattedAlgorithm = digestAlgorithm.toLowerCase().split('-').join('')
+
+  return { formattedAlgorithm, digestAlgorithm, digestValue }
+}
+
+const calculateDigest = ({ algorithm, message, encoding }) =>
```

Comment:
I think we have a method like this already. Should we look to see if it has been duplicated elsewhere?

Reply:
`constructSignatureString` has a `calculateSignature` but it uses `crypto.createHmac` instead of `crypto.createHash`

Reply:
I suppose I could refactor to allow method selection and then make this a single helper function. It looks like we only need this in two places so far. Is it worth it?

Reply:
Ahhh nope, thats sufficiently different :)

Reply:
This could also live directly in the getDigest method as well, but not required 👍